test-common/src/main/java/de/bwaldvogel/mongo/backend/AbstractBackendTest.java

```diff
@@ -250,6 +251,11 @@ public void testCountCommand() {
         assertThat(collection.count()).isZero();
     }
 
+    @Test
+    public void testNonPrimaryCountCommand() {
+        assertThat(collection.withReadPreference(ReadPreference.nearest()).countDocuments()).isZero();
```

Comment:
this gets the job done but isn't exactly descriptive. any recommendations for where else this change could be tested?

Reply:
What do you think about having two tests?
You could keep that test since a query with read preference was not yet covered.
However, the Java driver implementation could change anytime and thus it doesn’t guarantee to test what you want to test. So I would suggest to add another test where you construct a `$query` object explicitly?

Reply:
That seems like a good idea to me.